src/domain/gift-cards/new.tsx

```diff
@@ -0,0 +1,208 @@
+import { navigate } from "gatsby"
+import {
+  useAdminCreateProduct,
+  useAdminProducts,
+  useAdminStore,
+} from "medusa-react"
+import React, { useState } from "react"
+import { useForm } from "react-hook-form"
+import FileUploadField from "../../components/atoms/file-upload-field"
+import Button from "../../components/fundamentals/button"
+import PlusIcon from "../../components/fundamentals/icons/plus-icon"
+import TrashIcon from "../../components/fundamentals/icons/trash-icon"
+import InputField from "../../components/molecules/input"
+import Modal from "../../components/molecules/modal"
+import Textarea from "../../components/molecules/textarea"
+import CurrencyInput from "../../components/organisms/currency-input"
+import useToaster from "../../hooks/use-toaster"
+import { ProductStatus } from "../../types/shared"
+import { getErrorMessage } from "../../utils/error-messages"
+
+type NewGiftCardProps = {
+  onClose: () => void
+}
+
+type Denomination = {
+  name: string
+  component: React.ReactNode
+}
+
+const NewGiftCard: React.FC<NewGiftCardProps> = ({ onClose }) => {
+  const [uploading, setUploading] = useState(false)
+  const { register, setValue, unregister, handleSubmit } = useForm()
+  const { store } = useAdminStore()
+  const { refetch } = useAdminProducts()
+  const giftCard = useAdminCreateProduct()
+  const [denominations, setDenominations] = useState<Denomination[]>([])
+  const toaster = useToaster()
+
+  const handleValueUpdate = (name: string, value?: number) => {
+    if (!value) return
+    setValue(name, value)
+  }
+
+  const addDenomination = () => {
+    const name = `denominations.${denominations.length}`
+    const component = (
+      <CurrencyInput
+        currentCurrency={store?.default_currency_code}
+        readOnly
+        size="medium"
+      >
+        <CurrencyInput.AmountInput
+          label="Amount"
+          amount={undefined}
+          onChange={(v) => handleValueUpdate(name, v)}
+        />
+      </CurrencyInput>
+    )
+    register(name)
+    setDenominations([...denominations, { name, component }])
+  }
+
+  const deleteDenomination = (name: string) => {
+    unregister(name)
+    setDenominations(denominations.filter((d) => d.name !== name))
+  }
+
+  const handleFileUpload = async (files) => {
+    setUploading(true)
+    //TODO upload files
+    await new Promise((r) => setTimeout(r, 2000))
+    setUploading(false)
+  }
+
+  const onSubmit = (data: {
```

Comment:
There should be some data validation somewhere, this throws errors if it's submitted without any denominations